src/audio/buffer.c

```diff
@@ -251,10 +251,10 @@ void comp_update_buffer_consume(struct comp_buffer __sparse_cache *buffer, uint3
 	/* return if no bytes */
 	if (!bytes) {
 		buf_dbg(buffer, "comp_update_buffer_consume(), no bytes to consume, source->comp.id = %u, source->comp.type = %u, sink->comp.id = %u, sink->comp.type = %u",
-			dev_comp_id(buffer->source),
-			dev_comp_type(buffer->source),
-			dev_comp_id(buffer->sink),
-			dev_comp_type(buffer->sink));
+			buffer->source ? dev_comp_id(buffer->source) : (unsigned int)UINT32_MAX,
+			buffer->source ? dev_comp_type(buffer->source) : (unsigned int)UINT32_MAX,
+			buffer->sink ? dev_comp_id(buffer->sink) : (unsigned int)UINT32_MAX,
+			buffer->sink ? dev_comp_type(buffer->sink) : (unsigned int)UINT32_MAX);
```

Comment:
sorry, this is really a minor thing, this presumably anyway is "never" enabled, but why not just use `%d` and print `-1` if `NULL`? Would avoid casts huge numbers if this does get enabled and hit

Reply:
@lyakh Uhm,, dev_comp_id() is unsigned, so we need to use "%u". I could use "(unsigned int)-1", but to me UINT32_MAX seems better, more explicit. I don't get why it is a "long unsigned" to begin with (in some builds, but not all.

Reply:
Toolchain bug?

https://pubs.opengroup.org/onlinepubs/009696799/basedefs/stdint.h.html#tag_13_48_03_02

> ... this expression shall have the same type as would an expression that is an object of the corresponding type converted according to the integer promotions. Its implementation-defined value shall be equal to or greater in magnitude (absolute value) than the corresponding value given below, with the same sign, except where stated to be exactly the given value.

Good luck with that! Unless I really really have to, I usually quit at "integer promotion" :-)

Reply:
Alternative (TESTED)
```C
	/* return if no bytes */
	if (!bytes) {
		const uint32_t invalid = UINT32_MAX; // this is unsigned long in the Zephyr SDK
		buf_dbg(buffer, "comp_update_buffer_produce(), no bytes to produce, source->comp.id = %u, source->comp.type = %u, sink->comp.id = %u, sink->comp.type = %u",
			buffer->source ? dev_comp_id(buffer->source) : invalid,
			buffer->source ? dev_comp_type(buffer->source) : invalid,
			buffer->sink ? dev_comp_id(buffer->sink) : invalid,
			buffer->sink ? dev_comp_type(buffer->sink) : invalid);
		return;
	}
```

Reply:
> @lyakh Uhm,, dev_comp_id() is unsigned, so we need to use "%u". I could use "(unsigned int)-1", but to me UINT32_MAX seems better, more explicit. I don't get why it is a "long unsigned" to begin with (in some builds, but not all).

Would anything complain if you print an unsigned integer with the `%d` format? It didn't complain in my brief test locally

Reply:
@marc-hb make that `const unsigned int invalid` and I'd be ok with that :-)